### PR TITLE
#21 Build a REST API to retrieve a list of items

### PR DESCRIPTION
**Pull request by patsicko**
# What this PR do
## List available items for a buyer
# Description of the task
- Create a REST API to retrieve a list of all items in a seller's collection 
- Create a REST API to retrieve a list of available items to the buyer
## How should this be manually tested
- clone the repo
- use npm install to install all packages
- Add all required environment variables following the .env.example provided
- use npx sequelize db:migrate to migrate migrations
- Run npm run db:seeds to populate seeders in database
- use npm run dev to start the server
- Navigate to  the http://localhost:PORT/docs on your browser to access swagger APIs docs  by replacing 'PORT' with the port that your the app is listening to. 
- Create vendor role and permission.
- Sign up as a vendor, and login to have vendor's permissions
- Tryout GET//api/product/getAll and GET//api/product/getAvailable 
- A list of items will be displayed as a response

### CODE REVIEW

**Comment by patsicko** [comment]
[A user should be able to list all items](https://trello.com/c/fhatu25F/21-a-user-should-be-able-to-list-all-items)

**Comment by wakaflorien** [comment]
**Make a meaningfull PR description**

image

File: test/product.test.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  describe("testing get available items",() =>{
+    test('get available items', async () => { 
+        const res = await request(app).get("/api/product/getAvailable");
+        expect(res.statusCode).toBe(401);
+       
+     })
+  });
+
+
+  describe("testing get all items",() =>{
```
Comment (wakaflorien) [inline test/product.test.js:42]: - add more testcases.

**Review by wakaflorien — CHANGES_REQUESTED** [review]
- Correct and improve failing tests

**Review by sostene12 — COMMENTED** [review]
i'm having troubles in running creating products,

**Review by wakaflorien — APPROVED** [review]

**Review by oliviertech212 — APPROVED** [review]

**Review by wakaflorien — APPROVED** [review]

**Pull request merged**